colab/colab_utils.py

```diff
@@ -0,0 +1,1019 @@
+
+def progress(value, max=100):
+  css = """
+        <style>
+          progress {
+            border-radius: 7px;
+            box-shadow: 1px 1px 2px rgba(0, 0, 0, 0.5) inset;
+            width: 80%;
+            height: 30px;
+            display: block;
+          }
+          progress::-webkit-progress-bar {
+            background-color: rgba(237, 237, 237, 0);
+            border-radius: 7px;
+          }
+          progress::-webkit-progress-value {
+            background-color: green;
+            border-radius: 7px;
+            box-shadow: 1px 1px 1px rgba(0, 0, 0, 0.1) inset;
+          }
+        </style>
+        """
+  html = """
+          <progress
+              value='{value}'
+              max='{max}'
+          >
+            {value}%
+          </progress>
+        """.format(value=value, max=max)
+  return HTML(css + html)
+
+
+# Add custom basemaps to folium.
+
+basemaps = {
+    'Google Maps': folium.TileLayer(
+        tiles = 'https://mt1.google.com/vt/lyrs=m&x={x}&y={y}&z={z}',
+        attr = 'Google',
+        name = 'Google Maps',
+        overlay = True,
+        control = True
+    ),
+    'Google Satellite': folium.TileLayer(
+        tiles = 'https://mt1.google.com/vt/lyrs=s&x={x}&y={y}&z={z}',
+        attr = 'Google',
+        name = 'Google Satellite',
+        overlay = True,
+        control = True
+    ),
+    'Google Terrain': folium.TileLayer(
+        tiles = 'https://mt1.google.com/vt/lyrs=p&x={x}&y={y}&z={z}',
+        attr = 'Google',
+        name = 'Google Terrain',
+        overlay = True,
+        control = True
+    ),
+    'Google Satellite Hybrid': folium.TileLayer(
+        tiles = 'https://mt1.google.com/vt/lyrs=y&x={x}&y={y}&z={z}',
+        attr = 'Google',
+        name = 'Google Satellite',
+        overlay = True,
+        control = True
+    ),
+    'Esri Satellite': folium.TileLayer(
+        tiles = 'https://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{z}/{y}/{x}',
+        attr = 'Esri',
+        name = 'Esri Satellite',
+        overlay = True,
+        control = True
+    )
+}
+
+def create_folium_map_with_images(pathgcp_before, pathgcp_after):
+  # Load before image and get latitude/longitude of map center.
+
+  no_cog_file=''
+  error_message=[]
+  for image_path in pathgcp_before.split(',')+pathgcp_after.split(','):
+    try:
+      ee.Image.loadGeoTIFF(image_path).getMapId()
+    except Exception as e:
+      if re.search('The GeoTIFF is invalid or is not cloud optimized:',str(e)):
+        no_cog_file=f'{no_cog_file}{image_path}\n'
+        error_message.append(e)
+
+  if error_message:
+    print(f'The following TIFF image(s) need to be Cloud Optimzed GeoTIFF file(s) in order to be visualized in the map using EarthEngine:\n{no_cog_file}\n{str(error_message[0])}')
+    return
+
+  before_image_path = pathgcp_before.split(',')[0]
+  before_map = ee.Image.loadGeoTIFF(before_image_path)
+  before_map_id_dict = before_map.getMapId()
```

Comment:
This variable isn't used anywhere. Can get rid of this line.

Reply:
change commit 6d5256b